cmake/setup_compiler_flags_gnu.cmake

```diff
@@ -182,7 +182,13 @@ IF (CMAKE_BUILD_TYPE MATCHES "Debug")
   LIST(APPEND DEAL_II_DEFINITIONS_DEBUG "DEBUG")
   LIST(APPEND DEAL_II_USER_DEFINITIONS_DEBUG "DEBUG")
 
-  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS_DEBUG "-Og")
+  #
+  # In recent versions, gcc often eliminates too much debug information
+  # using '-Og' to be useful.
+  #
+  IF(NOT CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+    ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS_DEBUG "-Og")
+  ENDIF()
```

Comment:
Ah, sorry for being late to the show.

We can remove the statement altogether. Clang is the only other compiler recognizing `-Og` and for Clang it is synomymous to `-O1`.

Reply:
Do we want to use `-O0` rather than `-O1` for `clang` then? As far as I know the current debug symbols for `clang` are sufficient, aren't they?

Reply:
Actually, thinking about it a bit more - I think you're completely right. Let's keep `-Og` for clang. Maybe clang does something smarter in the future.